CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/TriviaPiecesFile.swift

```diff
@@ -264,4 +264,57 @@ let triviaPiecesFile = SourceFileSyntax(leadingTrivia: copyrightHeader) {
       }
     }
   }
+
+  try! generateIsHelpers(for: "TriviaPiece")
+
+  try! generateIsHelpers(for: "RawTriviaPiece")
+}
+
+fileprivate func generateIsHelpers(for pieceName: String) throws -> ExtensionDeclSyntax {
+  return try ExtensionDeclSyntax("extension \(raw: pieceName)") {
+    try VariableDeclSyntax("public var isBlank: Bool") {
+      try SwitchExprSyntax("switch self") {
+        for trivia in TRIVIAS {
+          if trivia.isBlank {
+            SwitchCaseSyntax("case .\(raw: trivia.enumCaseName):") {
+              StmtSyntax("return true")
+            }
+          }
+        }
+        SwitchCaseSyntax("default:") {
+          StmtSyntax("return false")
+        }
+      }
+    }
+
+    try VariableDeclSyntax("public var isNewline: Bool") {
+      try SwitchExprSyntax("switch self") {
+        for trivia in TRIVIAS {
+          if trivia.isNewLine {
+            SwitchCaseSyntax("case .\(raw: trivia.enumCaseName):") {
+              StmtSyntax("return true")
+            }
+          }
+        }
+        SwitchCaseSyntax("default:") {
+          StmtSyntax("return false")
+        }
+      }
+    }
+
+    DeclSyntax(
+      """
+      public var isIndentationWhitespace: Bool {
```

Comment:
I renamed these accessors and got rid of the term “blank” entirely because I found it ambiguous:
- `isNewline`: Obvious, though I have very little opinion on whether this should include form feeds and vertical tabs. I think it doesn’t make a difference in practice and if `Swift.String` considers them newlines, I don’t see a reason why we shouldn’t
- `isSpaceOrTab`: Let me know if you have a better name for it, I couldn’t find one and this is very clear IMO
- `isWhitespace`: `isSpaceOrTab || isNewline`